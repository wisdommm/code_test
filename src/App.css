@import '~antd/dist/antd.css';

.App {
  padding: 20px;
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.components-table-demo-nested .ant-table-expanded-row > td:last-child {
  padding: 0 48px 0 8px;
}

.components-table-demo-nested .ant-table-expanded-row > td:last-child .ant-table-thead th {
  border-bottom: 1px solid #e9e9e9;
}

.components-table-demo-nested .ant-table-expanded-row > td:last-child .ant-table-thead th:first-child {
  padding-left: 0;
}

.components-table-demo-nested .ant-table-expanded-row > td:last-child .ant-table-row td:first-child {
  padding-left: 0;
}

.components-table-demo-nested .ant-table-expanded-row .ant-table-row:last-child td {
  border: none;
}

.components-table-demo-nested .ant-table-expanded-row .ant-table-thead > tr > th {
  background: none;
}

.components-table-demo-nested .table-operation a:not(:last-child) {
  margin-right: 24px;
}

.components-table-demo-nested .ant-table-expanded-row:hover > td {
  background: #fbfbfb;
}

.summary_table{
  width: 100%;
  margin: 20px 10px;
  background-color: transparent;
  border-collapse: collapse;
  border-spacing: 0;
}

.summary_table thead{
  background: #fafafa;
  font-weight: 700;
}

.summary_table tr{
  width: 100%;
  vertical-align: top;
  text-align: left;
  font-size: 14px;
  border-bottom: 1px solid rgba(0, 0, 0, 0.15);
}

.summary_table td{
  width: 25%;
  padding: 16px;
}

table tr .key{
  font-weight: 600;
  color: #000;
}
