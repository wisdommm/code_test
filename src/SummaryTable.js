import React from 'react';
const SummaryTable = props => {
  let datas = props.datas || {};
  return <table className='summary_table'>
          <thead>
            <tr>
              <td>summary</td>
              <td></td>
              <td>hash</td>
              <td></td>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td className='key'>time</td>
              <td>{datas.time ? new Date(datas.time*1000).toLocaleString() : ''}</td>
              <td className='key'>hash</td>
              <td><div style={props.long_value_style}>{datas.hash || ''}</div></td>
            </tr>
            <tr>
              <td className='key'>nonce</td>
              <td>{datas.nonce || ''}</td>
              <td className='key'>prev_block</td>
              <td><div style={props.long_value_style}>{datas.prev_block || ''}</div></td>
            </tr>
            <tr>
              <td className='key'>size</td>
              <td>{datas.size || ''}</td>
              <td className='key'>next_block</td>
              <td><div style={props.long_value_style}>{datas.next_block || ''}</div></td>
            </tr>
            <tr>
              <td className='key'>bits</td>
              <td>{datas.bits || ''}</td>
              <td className='key'>mrkl_root</td>
              <td><div style={props.long_value_style}>{datas.mrkl_root || ''}</div></td>
            </tr>
            <tr>
              <td className='key'>ver</td>
              <td>{datas.ver || ''}</td>
              <td className='key'>n_tx</td>
              <td><div style={props.long_value_style}>{datas.n_tx || ''}</div></td>
            </tr>
          </tbody>
        </table>
}

export default SummaryTable;
