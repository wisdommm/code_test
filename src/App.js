import React, { Component } from 'react';
import { Button, Table, Input, Row, Col, notification, Switch } from 'antd';
import './App.css';

const math_add = (arg1, arg2) => {
    let r1, r2, m, c;
    try {
        r1 = arg1.toString().split(".")[1].length;
    }
    catch (e) {
        r1 = 0;
    }
    try {
        r2 = arg2.toString().split(".")[1].length;
    }
    catch (e) {
        r2 = 0;
    }
    c = Math.abs(r1 - r2);
    m = Math.pow(10, Math.max(r1, r2));
    if (c > 0) {
        let cm = Math.pow(10, c);
        if (r1 > r2) {
            arg1 = Number(arg1.toString().replace(".", ""));
            arg2 = Number(arg2.toString().replace(".", "")) * cm;
        } else {
            arg1 = Number(arg1.toString().replace(".", "")) * cm;
            arg2 = Number(arg2.toString().replace(".", ""));
        }
    } else {
        arg1 = Number(arg1.toString().replace(".", ""));
        arg2 = Number(arg2.toString().replace(".", ""));
    }
    return (arg1 + arg2) / m;
}

let long_value_style = {};

const columns = [
  { title: 'in', dataIndex: 'in', key: 'in',
    render: data => {
      return data && data.length > 0 ? data.map((e,i)=>{
        return <div style={long_value_style} key={i}>{e.coinbase || 'no data'}</div>
      }) : ''
    }
  },
  { title: 'out', dataIndex: 'out', key: 'out',
    render: data => {
      return data && data.length > 0 ? data.map((e,i)=>{
        return <div style={long_value_style} key={i}>{e.address || 'no data'}</div>
      }) : ''
    }
  },
  { title: 'count', dataIndex: 'out', key: 'count',
    render: data => {
      return data && data.length > 0 ? data.map((e,i)=>{
        return <div key={i}>{e.value || '0'}</div>
      }) : ''
    }
  },
  { title: 'total', dataIndex: 'out', key: 'total',
    render: data => {
      let total = 0;
      data && data.length > 0 ? data.map((e,i)=>{
        total = math_add( Number(e.value), total );
      }) : ''
      return <Button type='primary'>{total}</Button>
    }
  }
];

const columns_one = [
  { title: 'hash', dataIndex: 'hash', key: 'hash',
    render: data => {
      return <div style={long_value_style}>{data}</div>
    }
  },
  { title: 'size', dataIndex: 'size', key: 'size',
    render: data => {
      return <div>{data}</div>
    }
  },
  { title: 'nid', dataIndex: 'nid', key: 'nid',
    render: data => {
      return <div style={long_value_style}>{data}</div>
    }
  }
];

class App extends Component {
  constructor(){
    super(...arguments);
    this.state = {
      datas: [],
      showAddress: false
    };
    this.search = this.search.bind(this);
    this.showAddress = this.showAddress.bind(this);
  }

  componentDidMount(){
    this.search('000000000000000001806a922d4d35a37ad9324c690f72d556c6445cb7a9c214');
  }

  search(text){
    let _this = this;
    _this.setState({
      datas: {}
    });
    let url = `https://webbtc.com/block/${text}.json`;
    fetch( url ).then( json => {
      if(json.status === 200){
        json.text().then( responseText => {
          _this.setState({
            datas: JSON.parse(responseText)
          });
        });
      }else{
        notification.error({
          message: 'error',
          description: '请求失败'
        });
      }
    })
  }

  showAddress(show){
    this.setState({
      showAddress: show
    });
  }

  render() {
    const { datas, showAddress } = this.state;
    const expandedRowRender = () => {
      return <Table
              columns={columns}
              dataSource={datas.tx || []}
              pagination={true}
            />
    }

    long_value_style = showAddress ? { width: '500px' } : {
      width: '250px',
      whiteSpace: 'nowrap',
      overflow: 'hidden',
      textOverflow: 'ellipsis'
    }
    
    return (
      <div className="App">
        <Row>
          <Input.Search
            defaultValue='000000000000000001806a922d4d35a37ad9324c690f72d556c6445cb7a9c214'
            placeholder="input search text"
            onSearch={this.search}
            style={{ width: 300 }}
          />
        </Row>
        <Row style={{margin:'30px 0',textAlign: 'center'}}>
          是否展示完整地址：
          <Switch
            checkedChildren="是"
            unCheckedChildren="否"
            onChange={this.showAddress}
          />
        </Row>
        <Row style={{margin:'30px 0',textAlign: 'center'}}>
          <Col span={24}>
            <table className='summary_table'>
              <thead>
                <tr>
                  <td>summary</td>
                  <td></td>
                  <td>hash</td>
                  <td></td>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td className='key'>time</td>
                  <td>{datas.time ? new Date(datas.time*1000).toLocaleString() : ''}</td>
                  <td className='key'>hash</td>
                  <td><div style={long_value_style}>{datas.hash}</div></td>
                </tr>
                <tr>
                  <td className='key'>nonce</td>
                  <td>{datas.nonce || ''}</td>
                  <td className='key'>prev_block</td>
                  <td><div style={long_value_style}>{datas.prev_block}</div></td>
                </tr>
                <tr>
                  <td className='key'>size</td>
                  <td>{datas.size || ''}</td>
                  <td className='key'>next_block</td>
                  <td><div style={long_value_style}>{datas.next_block}</div></td>
                </tr>
                <tr>
                  <td className='key'>bits</td>
                  <td>{datas.bits || ''}</td>
                  <td className='key'>mrkl_root</td>
                  <td><div style={long_value_style}>{datas.mrkl_root}</div></td>
                </tr>
                <tr>
                  <td className='key'>ver</td>
                  <td>{datas.ver || ''}</td>
                  <td className='key'>n_tx</td>
                  <td><div style={long_value_style}>{datas.n_tx}</div></td>
                </tr>
              </tbody>
            </table>
          </Col>
          
        </Row>

        <Row>
          <Table
            className="components-table-demo-nested"
            columns={columns_one}
            dataSource={datas.tx || []}
            expandedRowRender={expandedRowRender}
            defaultExpandAllRows={true}
          />
        </Row>
      </div>
    );
  }
}

export default App;
