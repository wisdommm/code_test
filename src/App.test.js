import React from 'react';
import { shallow,mount } from 'enzyme';
import mock_data from './mock_data.js'
import getData from './fetch_data';
import App from './App';
import SummaryTable from './SummaryTable';
import { Table, Input, Switch } from 'antd';
const Search = Input.Search;

describe('<App />', () => {

	it('renders Components', () => {
	    const wrapper = shallow(<App />);

	    expect(wrapper.find(Table).length).toBe(1);

	    expect(wrapper.find(Search).length).toBe(1);

	    expect(wrapper.find(Switch).length).toBe(1);

	    expect(wrapper.find(SummaryTable).length).toBe(1);

	    expect( wrapper.find(SummaryTable).render().find('.key').length ).toBe(10);
	});

	it('should not display any info by default', () => {
	    const wrapper = shallow(<App />);

	    expect( JSON.stringify( wrapper.find('.components-table-demo-nested').prop('dataSource') ) ).toBe('[]');
	});

	it('should display a input to fill search text', () => {
	    const wrapper = mount(<App />);
	
	    expect(wrapper.find('input[type="text"]').length).toBe(1);
	});


	it('should render Components by default state', () => {
	    const wrapper = mount(<App />);

		expect( JSON.stringify( wrapper.state().datas ) ).toBe('{}');
	    
	    expect(wrapper.state().showAddress).toBe(false);
	});

	it('click Switch', () => {
		const wrapper = mount(<App />);

		expect( JSON.stringify( wrapper.find(Search).props().style ) ).toBe('{\"width\":300}');

		expect( JSON.stringify( wrapper.find(SummaryTable).props().long_value_style ) ).toBe('{\"width\":\"250px\",\"whiteSpace\":\"nowrap\",\"overflow\":\"hidden\",\"textOverflow\":\"ellipsis\"}');

		wrapper.find('.ant-switch').simulate('click');

		expect( JSON.stringify( wrapper.find(Search).props().style ) ).toBe('{\"width\":600}');

		expect( JSON.stringify( wrapper.find(SummaryTable).props().long_value_style ) ).toBe('{\"width\":\"500px\"}');
	});

	it('should render <App /> correctly',() => {
		const wrapper = shallow(<App />);

		expect(wrapper).toMatchSnapshot();

	});

	



	






});
