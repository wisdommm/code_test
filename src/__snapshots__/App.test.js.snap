// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render <App /> correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Row
          gutter={0}
>
          <Search
                    enterButton={false}
                    inputPrefixCls="ant-input"
                    onSearch={[Function]}
                    placeholder="input search text"
                    prefixCls="ant-input-search"
                    style={
                              Object {
                                        "width": 300,
                                      }
                    }
          />
</Row>,
        <Row
          gutter={0}
          style={
                    Object {
                              "margin": "30px 0",
                              "textAlign": "center",
                            }
          }
>
          是否展示完整地址：
          <Switch
                    checkedChildren="是"
                    onChange={[Function]}
                    prefixCls="ant-switch"
                    unCheckedChildren="否"
          />
</Row>,
        <Row
          gutter={0}
          style={
                    Object {
                              "margin": "30px 0",
                              "textAlign": "center",
                            }
          }
>
          <Col
                    span={24}
          >
                    <SummaryTable
                              class="SummaryTable"
                              datas={Object {}}
                              long_value_style={
                                        Object {
                                                  "overflow": "hidden",
                                                  "textOverflow": "ellipsis",
                                                  "whiteSpace": "nowrap",
                                                  "width": "250px",
                                                }
                              }
                    />
          </Col>
</Row>,
        <Row
          gutter={0}
>
          <Table
                    bordered={false}
                    className="components-table-demo-nested"
                    columns={
                              Array [
                                        Object {
                                          "dataIndex": "hash",
                                          "key": "hash",
                                          "render": [Function],
                                          "title": "hash",
                                        },
                                        Object {
                                          "dataIndex": "size",
                                          "key": "size",
                                          "render": [Function],
                                          "title": "size",
                                        },
                                        Object {
                                          "dataIndex": "nid",
                                          "key": "nid",
                                          "render": [Function],
                                          "title": "nid",
                                        },
                                      ]
                    }
                    dataSource={Array []}
                    defaultExpandAllRows={true}
                    expandedRowRender={[Function]}
                    indentSize={20}
                    loading={false}
                    locale={Object {}}
                    prefixCls="ant-table"
                    rowKey="key"
                    showHeader={true}
                    size="large"
                    useFixedHeader={false}
          />
</Row>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Search
            enterButton={false}
            inputPrefixCls="ant-input"
            onSearch={[Function]}
            placeholder="input search text"
            prefixCls="ant-input-search"
            style={
                        Object {
                                    "width": 300,
                                  }
            }
/>,
          "gutter": 0,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "enterButton": false,
            "inputPrefixCls": "ant-input",
            "onSearch": [Function],
            "placeholder": "input search text",
            "prefixCls": "ant-input-search",
            "style": Object {
              "width": 300,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "是否展示完整地址：",
            <Switch
              checkedChildren="是"
              onChange={[Function]}
              prefixCls="ant-switch"
              unCheckedChildren="否"
/>,
          ],
          "gutter": 0,
          "style": Object {
            "margin": "30px 0",
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": Array [
          "是否展示完整地址：",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checkedChildren": "是",
              "onChange": [Function],
              "prefixCls": "ant-switch",
              "unCheckedChildren": "否",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Col
            span={24}
>
            <SummaryTable
                        class="SummaryTable"
                        datas={Object {}}
                        long_value_style={
                                    Object {
                                                "overflow": "hidden",
                                                "textOverflow": "ellipsis",
                                                "whiteSpace": "nowrap",
                                                "width": "250px",
                                              }
                        }
            />
</Col>,
          "gutter": 0,
          "style": Object {
            "margin": "30px 0",
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <SummaryTable
              class="SummaryTable"
              datas={Object {}}
              long_value_style={
                            Object {
                                          "overflow": "hidden",
                                          "textOverflow": "ellipsis",
                                          "whiteSpace": "nowrap",
                                          "width": "250px",
                                        }
              }
/>,
            "span": 24,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "class": "SummaryTable",
              "datas": Object {},
              "long_value_style": Object {
                "overflow": "hidden",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
                "width": "250px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Table
            bordered={false}
            className="components-table-demo-nested"
            columns={
                        Array [
                                    Object {
                                      "dataIndex": "hash",
                                      "key": "hash",
                                      "render": [Function],
                                      "title": "hash",
                                    },
                                    Object {
                                      "dataIndex": "size",
                                      "key": "size",
                                      "render": [Function],
                                      "title": "size",
                                    },
                                    Object {
                                      "dataIndex": "nid",
                                      "key": "nid",
                                      "render": [Function],
                                      "title": "nid",
                                    },
                                  ]
            }
            dataSource={Array []}
            defaultExpandAllRows={true}
            expandedRowRender={[Function]}
            indentSize={20}
            loading={false}
            locale={Object {}}
            prefixCls="ant-table"
            rowKey="key"
            showHeader={true}
            size="large"
            useFixedHeader={false}
/>,
          "gutter": 0,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bordered": false,
            "className": "components-table-demo-nested",
            "columns": Array [
              Object {
                "dataIndex": "hash",
                "key": "hash",
                "render": [Function],
                "title": "hash",
              },
              Object {
                "dataIndex": "size",
                "key": "size",
                "render": [Function],
                "title": "size",
              },
              Object {
                "dataIndex": "nid",
                "key": "nid",
                "render": [Function],
                "title": "nid",
              },
            ],
            "dataSource": Array [],
            "defaultExpandAllRows": true,
            "expandedRowRender": [Function],
            "indentSize": 20,
            "loading": false,
            "locale": Object {},
            "prefixCls": "ant-table",
            "rowKey": "key",
            "showHeader": true,
            "size": "large",
            "useFixedHeader": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Row
            gutter={0}
>
            <Search
                        enterButton={false}
                        inputPrefixCls="ant-input"
                        onSearch={[Function]}
                        placeholder="input search text"
                        prefixCls="ant-input-search"
                        style={
                                    Object {
                                                "width": 300,
                                              }
                        }
            />
</Row>,
          <Row
            gutter={0}
            style={
                        Object {
                                    "margin": "30px 0",
                                    "textAlign": "center",
                                  }
            }
>
            是否展示完整地址：
            <Switch
                        checkedChildren="是"
                        onChange={[Function]}
                        prefixCls="ant-switch"
                        unCheckedChildren="否"
            />
</Row>,
          <Row
            gutter={0}
            style={
                        Object {
                                    "margin": "30px 0",
                                    "textAlign": "center",
                                  }
            }
>
            <Col
                        span={24}
            >
                        <SummaryTable
                                    class="SummaryTable"
                                    datas={Object {}}
                                    long_value_style={
                                                Object {
                                                            "overflow": "hidden",
                                                            "textOverflow": "ellipsis",
                                                            "whiteSpace": "nowrap",
                                                            "width": "250px",
                                                          }
                                    }
                        />
            </Col>
</Row>,
          <Row
            gutter={0}
>
            <Table
                        bordered={false}
                        className="components-table-demo-nested"
                        columns={
                                    Array [
                                                Object {
                                                  "dataIndex": "hash",
                                                  "key": "hash",
                                                  "render": [Function],
                                                  "title": "hash",
                                                },
                                                Object {
                                                  "dataIndex": "size",
                                                  "key": "size",
                                                  "render": [Function],
                                                  "title": "size",
                                                },
                                                Object {
                                                  "dataIndex": "nid",
                                                  "key": "nid",
                                                  "render": [Function],
                                                  "title": "nid",
                                                },
                                              ]
                        }
                        dataSource={Array []}
                        defaultExpandAllRows={true}
                        expandedRowRender={[Function]}
                        indentSize={20}
                        loading={false}
                        locale={Object {}}
                        prefixCls="ant-table"
                        rowKey="key"
                        showHeader={true}
                        size="large"
                        useFixedHeader={false}
            />
</Row>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Search
              enterButton={false}
              inputPrefixCls="ant-input"
              onSearch={[Function]}
              placeholder="input search text"
              prefixCls="ant-input-search"
              style={
                            Object {
                                          "width": 300,
                                        }
              }
/>,
            "gutter": 0,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "enterButton": false,
              "inputPrefixCls": "ant-input",
              "onSearch": [Function],
              "placeholder": "input search text",
              "prefixCls": "ant-input-search",
              "style": Object {
                "width": 300,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "是否展示完整地址：",
              <Switch
                checkedChildren="是"
                onChange={[Function]}
                prefixCls="ant-switch"
                unCheckedChildren="否"
/>,
            ],
            "gutter": 0,
            "style": Object {
              "margin": "30px 0",
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            "是否展示完整地址：",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "checkedChildren": "是",
                "onChange": [Function],
                "prefixCls": "ant-switch",
                "unCheckedChildren": "否",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Col
              span={24}
>
              <SummaryTable
                            class="SummaryTable"
                            datas={Object {}}
                            long_value_style={
                                          Object {
                                                        "overflow": "hidden",
                                                        "textOverflow": "ellipsis",
                                                        "whiteSpace": "nowrap",
                                                        "width": "250px",
                                                      }
                            }
              />
</Col>,
            "gutter": 0,
            "style": Object {
              "margin": "30px 0",
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <SummaryTable
                class="SummaryTable"
                datas={Object {}}
                long_value_style={
                                Object {
                                                "overflow": "hidden",
                                                "textOverflow": "ellipsis",
                                                "whiteSpace": "nowrap",
                                                "width": "250px",
                                              }
                }
/>,
              "span": 24,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "class": "SummaryTable",
                "datas": Object {},
                "long_value_style": Object {
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                  "width": "250px",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Table
              bordered={false}
              className="components-table-demo-nested"
              columns={
                            Array [
                                          Object {
                                            "dataIndex": "hash",
                                            "key": "hash",
                                            "render": [Function],
                                            "title": "hash",
                                          },
                                          Object {
                                            "dataIndex": "size",
                                            "key": "size",
                                            "render": [Function],
                                            "title": "size",
                                          },
                                          Object {
                                            "dataIndex": "nid",
                                            "key": "nid",
                                            "render": [Function],
                                            "title": "nid",
                                          },
                                        ]
              }
              dataSource={Array []}
              defaultExpandAllRows={true}
              expandedRowRender={[Function]}
              indentSize={20}
              loading={false}
              locale={Object {}}
              prefixCls="ant-table"
              rowKey="key"
              showHeader={true}
              size="large"
              useFixedHeader={false}
/>,
            "gutter": 0,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bordered": false,
              "className": "components-table-demo-nested",
              "columns": Array [
                Object {
                  "dataIndex": "hash",
                  "key": "hash",
                  "render": [Function],
                  "title": "hash",
                },
                Object {
                  "dataIndex": "size",
                  "key": "size",
                  "render": [Function],
                  "title": "size",
                },
                Object {
                  "dataIndex": "nid",
                  "key": "nid",
                  "render": [Function],
                  "title": "nid",
                },
              ],
              "dataSource": Array [],
              "defaultExpandAllRows": true,
              "expandedRowRender": [Function],
              "indentSize": 20,
              "loading": false,
              "locale": Object {},
              "prefixCls": "ant-table",
              "rowKey": "key",
              "showHeader": true,
              "size": "large",
              "useFixedHeader": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
